src/components/ui/sidebar/nav-main.tsx

```diff
@@ -46,19 +46,7 @@ export interface NavigationLink {
 export function NavMain({ links }: { links: NavigationLink[] }) {
   const { state } = useSidebar();
   const isCollapsed = state === "collapsed";
-
   const fullPath = useFullPath();
-  const fullPathMap = useMemo(
-    () =>
-      fullPath.split("/").reduce(
```

Comment:
Removing this breaks existing functionality (when removing existing code, always ask the qn: what purpose does it serve?).

In this case, this is required for urls like /encounters/id/updates, /encounters/id/plots, patient/id/demography etc.